sklearn/preprocessing/_polynomial.py

```diff
@@ -750,8 +762,15 @@ def _get_base_knot_positions(X, n_knots=10, knots="uniform", sample_weight=None)
             )
 
             if sample_weight is None:
-                knots = np.percentile(X, percentiles, axis=0)
+                knots = np.nanpercentile(X, percentiles, axis=0)
```

Comment:
It's possible that might need to replace `np.nan` by 0.0 to be able to encode missing values in all-nan columns without raising an error.

Since any constant column (all percentiles are the same) will be spline encoded to 0.0, it does not matter which value we use to represent the constant knots values derived from an all-nan training data column.

Note that I am not 100% sure whether this is needed or not.